### ISSUE TITLE
[Self-managed]: 7.13.1 agent stuck in Updating state on triggering Upgrade on 7.14.0 Snapshot

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/beats#26133]
**Kibana version:** 7.14.0 Snapshot upgraded from 7.13.0 self-managed Kibana environment

**Host OS and Browser version:** Windows 10 x64, All

**Build Details:**
```
  Artifact link used[7.13.1]: https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-7.13.1-windows-x86_64.zip
  Build: 41403
  Commit: 4f0781737cdb5cf9b5dfa624368b4306b5d6bcb0
```

**Preconditions:**
1. Upgraded 7.14.0 Snapshot  self-managed  Kibana environment should be available.
2. 7.13.1 released Windows 10 x64 Fleet Server Agent must be installed with policy having System, Endpoint Security and Fleet Server integration.

**Steps to reproduce:**
1. Login to Kibana environment.
2. After Kibana upgraded to 7.14.0, upgrade existing 7.13.1 Fleet Server Agent to 7.14.0
3. Observe Agent stuck in Updating state and didn't get upgraded.

**Expected Result:**
Agent should be upgraded to 7.14.0 and should be Healthy on Kibana UI.

**Note:**
- 7.13.1 agent upgraded to 7.14.0 on 7.14.0 Snapshot Kibana Cloud (Staging).

**Screenshot:**
![8](https://user-images.githubusercontent.com/77374876/120648796-aa3b8580-c499-11eb-9d03-d6957656589a.JPG)

**Comment by amolnater-qasource** [comment]
@karanbirsingh-qasource Please review.

**Comment by EricDavisX** [comment]
we are tracking this problem here, unless there is a need to keep this open (I don't think there is) - we can close this.  @amolnater-qasource let's use just the one issue for upgrade problems until we get them resolved, we have a few duplicates open:
https://github.com/elastic/beats/issues/26041

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX

Thanks for the feedback.

However, this issue was reported being specific to self managed environment.

We will retest this issue once https://github.com/elastic/beats/issues/26041 will be fixed.

Thanks
QAS

**Comment by amolnater-qasource** [comment]
Hi @EricDavisX 

We have revalidated this issue on 7.14.0 self-managed Snapshot Kibana and found it **fixed**.

**Build Details:**
```
Artifact link: https://artifacts.elastic.co/downloads/beats/elastic-agent/elastic-agent-7.13.1-windows-x86_64.zip
Build: 41559
Commit: 9838db392e7fcfc12f004b68fb1b09739f131148
```

**What's working:**
We are now able to upgrade 7.13.1(released) agent to 7.14.0 on self managed 7.14.0 Snapshot Kibana.

Thanks
QAS
